pkg/spoke/auth/auth.go

```diff
@@ -91,14 +81,9 @@ func (v *sarValidator) Validate(ctx context.Context, executor *workapiv1.Manifes
 		return fmt.Errorf("the executor service account is nil")
 	}
 
-	var verbs []string
-	switch action {
-	case ApplyAction:
-		verbs = []string{"create", "update", "patch", "get"}
-	case DeleteAction:
-		verbs = []string{"delete"}
-	default:
-		return fmt.Errorf("execute action %s is invalid", action)
+	verbs := []string{"create", "update", "patch", "get"}
+	if ownedByTheWork {
```

Comment:
we should add a TODO here indicating the potential risk.

Reply:
added some comments, ptal